### PR TITLE
[IGNORE] Playwright Dev Testing & CI Setup

### PR DESCRIPTION
**Pull request by ashra-f**
### Purpose

This PR improves application reliability by implementing testing with Playwright and integrating it into the CI pipeline. It detects issues early and ensures consistent test execution for early detection of bugs and breaking changes.

The key actions taken to achieve this are:

1. **Playwright Setup**: We've configured Playwright as our testing framework. It emulates user interactions, thereby helping us test our application more comprehensively.
2. **Creating Basic Tests**: To validate our setup and provide a blueprint for future tests, we've created a set of basic tests.
3. **CI Integration**: We've updated our CI pipeline to automatically execute our Playwright tests on each push to the main branch and with each new PR. This keeps our codebase healthy by ensuring that all changes are validated against our test suite before integration.

Here's a [screenshot](https://imgur.com/a/ex3mLIb) demonstrating the successful execution of our tests.

This is the initial step in creating a robust test suite. As our application evolves, we will continue expanding our tests to cover new features and edge cases. We welcome any suggestions or improvements to enhance our current setup.

### CODE REVIEW

**Comment by ashra-f** [comment]
It turns out that the problem with the Cloudflare Pages deployment was related to the Node.js version not meeting the minimum requirement specified by @astrojs/react@2.2.1, which needed a version of Node.js that was at least 16.12.0. Thanks to a helpful individual in the Cloudflare community, the issue was [resolved](https://community.cloudflare.com/t/pages-node-js-version/295548/2).

**Comment by ashra-f** [comment]
the tests seem to be running twice on every push to the main branch.

**Comment by cheningD** [comment]
> the tests seem to be running twice on every push to the main branch.

I think this is because we have a trigger for every push and for every PR.
If we are running playwright tests on every push that covers PR's automatically.

File: .github/workflows/playwright.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Playwright Tests
+on:
+  push:
+  pull_request:
+    branches: [main, master]
```
Comment (cheningD) [inline .github/workflows/playwright.yml:5]: We only need to run tests on push, because every PR will be a push so having it as a separate option is not necessary i think.
Also we don't need to include 'master' because we only use 'main' in this repo

File: .github/workflows/playwright.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+jobs:
+  test:
+    timeout-minutes: 60
```
Comment (cheningD) [inline .github/workflows/playwright.yml:8]: Let's use 5 minutes as a timeout. If the tests take longer than that for the MVP, our tests are too slow!

File: README.md [hunk 3]
```diff
@@ -38,14 +38,121 @@ Any static assets, like images, can be placed in the `public/` directory.
...
+yarn create playwright
+```
+
+Since we are using Playwright for all testing, we need to further install Playwright Test for [components](https://playwright.dev/docs/test-components) for our respective framework (React, Vue, Svelte, etc.). For example, if you are using React, you would run:
+
+```
+npm init playwright@latest -- --ct
+```
+
+Note: We need to set the test directory to `tests/frontend/unit` to only run unit tests for our frontend.
```
Comment (cheningD) [inline README.md:99]: How would you do this (e.g. mention the where the change must be made)? When would we want to do this?

**Review by cheningD — APPROVED** [review]
Thank you for the examples!
Just a few nits that can be updated before merge
